prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -309,6 +314,58 @@ function installKyma() {
     "${KYMA_SCRIPTS_DIR}"/is-installed.sh --timeout ${KYMA_INSTALL_TIMEOUT}
 }
 
+function checkTestPodTerminated() {
+    local retry=0
+    local runningPods=0
+    local completedPods=0
+
+    while [ "${retry}" -lt "${UPGRADE_TEST_TIMEOUT}" ]; do
+        for podName in $(kubectl get pods -n "${UPGRADE_TEST_NAMESPACE}" -l "${UPGRADE_TEST_RESOURCE_LABEL}" -o json | jq -sr '.[]|.items[].metadata.name')
+        do
+            runningPods=$((runningPods + 1))
+            phase=$(kubectl get pod "${podName}" -n "${UPGRADE_TEST_NAMESPACE}" -o json | jq '.status.phase')
+
+            if [[ "${phase}" == *"Succeeded"* ]] || [[ "${phase}" == *"Failed"* ]]
+            then
+              echo "Test pod '${podName}' has not terminated, pod phase: ${phase}"
+              completedPods=$((completedPods + 1))
+            fi
+        done
+
+        if [ "${runningPods}" == "${completedPods}" ]
+        then
+            echo "All pods in ${UPGRADE_TEST_NAMESPACE} namespace have not terminated phase"
+            return 1
+        fi
+
+        echo "Not all pods in ${UPGRADE_TEST_NAMESPACE} namespace have not terminated phase. Retry checking."
+        runningPods=0
+        completedPods=0
+        retry=$((retry + 1))
+        sleep 1
+    done
+}
+
+createTestResources() {
+    shout "Create test e2e resources"
+    date
+
+    release=$(basename "${UPGRADE_TEST_PATH}")
+    export release
+
+    helm install "${UPGRADE_TEST_PATH}" \
+        --name "${release}" \
+        --namespace "${UPGRADE_TEST_NAMESPACE}" \
+        --timeout "${UPGRADE_TEST_HELM_TIMEOUT}" \
+        --wait
+    prepareResult=$?
+
+    checkTestPodTerminated
+    echo "Exit status for prepare upgrade e2e tests: ${prepareResult}"
+    echo "Test pods logs: "
+    kubectl logs -l "${UPGRADE_TEST_RESOURCE_LABEL}" | jq -s '.[]|.log|select(.message | contains("CreateResources")) |.message'
```

Comment:
I see a problem right now. We need to have different labels for pod which is creating resources and testing them. IMO the `contains("CreateResources")` is not acceptable because sth we log more general information e.g. app cannot start at all because of wrong configuration 


My suggestion:

- "kyma-project.io/upgrade-e2e-test": "prepareData" for pod which is creating data
- "kyma-project.io/upgrade-e2e-test": "executeTests" for pods which is executing tests